src/shims/unix/socket.rs

```diff
@@ -48,6 +103,8 @@ impl FileDescription for SocketPair {
         if let Some(writebuf) = self.writebuf.upgrade() {
             writebuf.borrow_mut().buf_has_writer = false;
         };
+        // TODO: how to notify another file description from here? We only have access to
+        // current file description.
```

Comment:
Instead of a `buf_has_writer` field, you could store a `WeakFileDescriptor` and modify the other socketpair socket through that